### ISSUE TITLE
Customizing one child class impacts other child classes

### ISSUE THREAD

**Issue by ploeh** [AutoFixture/AutoFixture#531]
Consider this test code:

``` C#
public class MyBase
{
    public string Id { get; set; }
}

public class ChildA : MyBase { }

public class ChildB : MyBase { }

[Fact]
public void Repro()
{
    var fixture = new Fixture();
    fixture.Customize<ChildA>(c => c.With(x => x.Id, "foo"));

    var b = fixture.Create<ChildB>();

    Assert.NotNull(b.Id); // This assertion fails
}
```

Surprisingly, as the comment explains, the test fails.

As far as I can tell, this happens because [the implementation of `With`](https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoFixture/Dsl/NodeComposer.cs#L337-L368) creates an `Omitter` with a `MemberInfoEqualityComparer`.

`MemberInfoEqualityComparer` uses `DeclaringType` to compare two members, and since `Id` is declared on `MyBase`, this impacts all `MyBase.Id` properties, while it should have impacted only `ChildA.Id`.

While it'd seem that the most straightforward fix would be to instead use `ReflectedType`, apparently [this property isn't going to be in .NET Core](https://github.com/dotnet/coreclr/issues/926#issuecomment-101032799), so for the sake of future compatibility, we should look for other options.

**Comment by ploeh** [comment]
BTW, this bug may or may not be related to #321

**Comment by ploeh** [comment]
This issue is a condensed version of http://stackoverflow.com/q/35487256/126014

**Comment by zvirja** [comment]
Fixed in #781. Will be released as a part of v4.
